url73e3057d61_empeeoh_BACnet4J_tgz-pJ8-com_serotonin_bacnet4j_test_OverflowTestJ8/src/com/serotonin/bacnet4j/npdu/mstp/MstpNode.java

```diff
@@ -0,0 +1,610 @@
+    }
+
+    public void setNetwork(MstpNetwork network) {
+        this.network = network;
+    }
+
+    /**
+     * @param timeSource
+     *            the timeSource to set
+     */
+    public void setTimeSource(TimeSource timeSource) {
+        this.timeSource = timeSource;
+    }
+
+    /**
+     * The flag ReceiveError is TRUE if an error is detected during the reception of an octet. Many common UARTs detect
+     * several types of receive errors, in particular framing errors and overrun errors. ReceiveError shall be TRUE if
+     * any of these errors is detected.
+     */
+    private boolean receiveError;
+
+    /**
+     * Substitute for the silence timer.
+     */
+    protected long lastNonSilence;
+
+    /**
+     * Used to store the data on the incoming frame.
+     */
+    protected final Frame frame = new Frame();
+
+    /**
+     * The header CRC accumulator.
+     */
+    private final HeaderCRC headerCRC = new HeaderCRC();
+
+    /**
+     * The data CRC accumulator.
+     */
+    private final DataCRC dataCRC = new DataCRC();
+
+    /**
+     * Used as an index by the Receive State Machine, up to a maximum value of InputBufferSize.
+     */
+    private int index;
+
+    /**
+     * Used to count the number of received octets or errors. This is used in the detection of link activity.
+     */
+    int eventCount;
+
+    /**
+     * The number of frames sent by this node during a single token hold. When this counter reaches the value
+     * Nmax_info_frames, the node must pass the token.
+     */
+    protected int frameCount;
+
+    /**
+     * An array of octets, used to store octets as they are received. InputBuffer is indexed from 0 to
+     * InputBufferSize-1. The maximum size of a frame is 501 octets. A smaller value for InputBufferSize may be used by
+     * some implementations.
+     */
+    private final ByteQueue inputBuffer = new ByteQueue(MAX_FRAME_LENGTH);
+
+    /**
+     * ReceivedInvalidFrame A Boolean flag set to TRUE by the Receive State Machine if an error is detected during the
+     * reception of a frame. Set to FALSE by the main state machine.
+     */
+    protected String receivedInvalidFrame;
+
+    /**
+     * A Boolean flag set to TRUE by the Receive State Machine if a valid frame is received. Set to FALSE by the main
+     * state machine.
+     */
+    protected boolean receivedValidFrame;
+
+    protected boolean activity;
+
+    /**
+     * @return the running
+     */
+    public boolean isRunning() {
+        return running;
+    }
+
+    /**
+     * @return the thisStation
+     */
+    public byte getThisStation() {
+        return thisStation;
+    }
+
+    @Override
+    public final void run() {
+        while (running) {
+            activity = false;
+
+            doCycle();
+
+            if (!activity && inactivityDelay > 0) {
+                try {
+                    Thread.sleep(inactivityDelay);
+                }
+                catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)